website/docs/r/globalaccelerator_endpoint_group.html.markdown

```diff
@@ -39,8 +39,10 @@ The following arguments are supported:
 
 **endpoint_configuration** supports the following attributes:
 
+* `client_ip_preservation_enabled` - (Optional) Indicates whether client IP address preservation is enabled for an Application Load Balancer endpoint. See the [AWS documentation](https://docs.aws.amazon.com/global-accelerator/latest/dg/preserve-client-ip-address.html) for more details.
+**Note:** when client IP address preservation is enabled, the Global Accelerator service creates a security group that is not deleted when the accelerator is deleted.
```

Comment:
😭  this is why we cannot have nice things! (not this specifically, the service behavior). I'm fairly concerned that we will need to add a lot more documentation around what to do with these if we are not going to provide a configuration way to remove them such as:

```
# Design sketch for potential enhancement.
# This is intended for discussion, not suggestive of a future implementation.
resource "aws_vpc" "example" {
  # ... other configuration ...

  force_destroy_global_accelerator_security_group = true
}
```

That documentation could be something like providing the AWS CLI commands or as simple as providing the group name here to start.

```suggestion
**Note:** When client IP address preservation is enabled, the Global Accelerator service creates an EC2 Security Group in the VPC named `GlobalAccelerator` that must be deleted (potentially outside of Terraform) before the VPC will successfully delete. If this EC2 Security Group is not deleted, Terraform will retry the VPC deletion for a few minutes before reporting a `DependencyViolation` error. This cannot be resolved by re-running Terraform.
```